framework/include/materials/MaterialData.h

```diff
@@ -296,19 +358,42 @@ MaterialData::getProperty(const std::string & name)
   resizeProps<T>(prop_id);
   auto prop = dynamic_cast<MaterialProperty<T> *>(_props[prop_id]);
   if (!prop)
-    mooseError("Material has no property named: " + name);
+  {
+    // We didn't find a regular material property so we're going to error out. But we can check to
+    // see whether there is an AD property of the same name in the hope that we can give the user a
+    // more meaningful error message
+    auto ad_prop = dynamic_cast<ADMaterialProperty<T> *>(_props[prop_id]);
+    if (ad_prop)
+      mooseError("The requested regular material property " + name +
+                 " is declared as an AD property. Either retrieve it as an AD property with "
+                 "getADMaterialProperty or declare it as a regular property wtih declareProperty");
+    else
+      mooseError("Material has no property named: " + name);
```

Comment:
What was the motivation behind removing graceful fallback from AD to normal properties?  It was definitely useful - especially when incrementally converting apps over to AD.

It's really too bad our material API doesn't have users calling functions at the use-point - then it would be trivial to gracefully throw away derivatives to provide a regular property value when an AD one was defined. Or throw warnings, etc.

Reply:
I think this comment is the most forefront illustration: https://github.com/idaholab/moose/pull/14701#issuecomment-605265668.

I also like @joshuahansel's comment here: https://github.com/idaholab/moose/pull/14701#issuecomment-606153585

Reply:
Yeah - the interoperability was very confusing and multiple times was the source of major problems - probably needed to go.  It's really too bad we don't have function calls to tap into - this and so many other problems would just be complete non-issues.

Reply:
That's very true. If a user wants to explicitly throw away things (or even try to add derivatives?) they should be able to. Perhaps that's something we can add later